src/proposals/mips/mip-b21/mip-b21.sol

```diff
@@ -0,0 +1,220 @@
+
+    /// @notice metamorpho storage slot offset for pending owner
+    bytes32 public constant PENDING_OWNER_SLOT = bytes32(uint256(9));
+
+    constructor() {
+        bytes memory proposalDescription = abi.encodePacked(
+            vm.readFile("./src/proposals/mips/mip-b21/MIP-B21.md")
+        );
+
+        _setProposalDescription(proposalDescription);
+    }
+
+    /// @notice proposal's actions all happen on base
+    function primaryForkId() public view override returns (uint256) {
+        return baseForkId;
+    }
+
+    function deploy(Addresses addresses, address) public override {}
+
+    function afterDeploy(Addresses addresses, address) public override {}
+
+    function afterDeploySetup(Addresses addresses) public override {
+        address usdcMetaMorphoVault = addresses.getAddress(
+            "USDC_METAMORPHO_VAULT"
+        );
+        address wethMetaMorphoVault = addresses.getAddress(
+            "WETH_METAMORPHO_VAULT"
+        );
+        address temporalGovernor = addresses.getAddress("TEMPORAL_GOVERNOR");
+        bytes32 pendingOwner = bytes32(uint256(uint160(temporalGovernor)));
+
+        if (
+            Ownable2StepUpgradeable(usdcMetaMorphoVault).pendingOwner() !=
+            temporalGovernor
+        ) {
+            vm.store(usdcMetaMorphoVault, PENDING_OWNER_SLOT, pendingOwner);
+        }
+        if (
+            Ownable2StepUpgradeable(wethMetaMorphoVault).pendingOwner() !=
+            temporalGovernor
+        ) {
+            vm.store(wethMetaMorphoVault, PENDING_OWNER_SLOT, pendingOwner);
+        }
+    }
+
+    function build(Addresses addresses) public override {
+        address foundationMultisig = addresses.getAddress(
+            "FOUNDATION_MULTISIG"
+        );
+        address temporalGovernor = addresses.getAddress("TEMPORAL_GOVERNOR");
+        ERC20Upgradeable well = ERC20Upgradeable(
+            addresses.getAddress("xWELL_PROXY")
+        );
+
+        if (
+            well.allowance(foundationMultisig, temporalGovernor) < WELL_AMOUNT
+        ) {
+            vm.prank(foundationMultisig);
+            well.approve(temporalGovernor, WELL_AMOUNT);
+        }
+
+        if (well.balanceOf(foundationMultisig) < WELL_AMOUNT) {
+            deal(address(well), foundationMultisig, WELL_AMOUNT);
+        }
+
+        _pushCrossChainAction(
+            addresses.getAddress("USDC_METAMORPHO_VAULT"),
+            abi.encodeWithSignature("acceptOwnership()"),
+            "Accept ownership of the Moonwell USDC Metamorpho Vault"
+        );
+        _pushCrossChainAction(
+            addresses.getAddress("WETH_METAMORPHO_VAULT"),
+            abi.encodeWithSignature("acceptOwnership()"),
+            "Accept ownership of the Moonwell WETH Metamorpho Vault"
+        );
+        _pushCrossChainAction(
+            addresses.getAddress("xWELL_PROXY"),
+            abi.encodeWithSignature(
+                "transferFrom(address,address,uint256)",
```

Comment:
when did the approval occur? Is the `FOUNDATION_MULTISIG` the sender of the tx?

Reply:
@MerlinEgalite after the KYC/KYB process is completed by Morpho and the Moonwell Foundation directors approve the grant, the approval will be opened in order for the proposal to properly execute.

Reply:
very clear thx